apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx

```diff
@@ -133,26 +141,101 @@ const getNativeElementsList = (elements: SBEnumType['value']): JSX.Element => {
   );
 };
 
-const RenderArgsTable = ({ hideArgsTable, primaryStory }: { primaryStory: PrimaryStory; hideArgsTable: boolean }) => {
+const slotRegex = /as\?:\s*"([^"]+)"/;
+function withSlotEnhancer(story: PreparedStory) {
+  const updatedArgTypes = { ...story.argTypes };
+  const hasArgAsProp = story.argTypes.as && story.argTypes.as?.type?.name === 'enum';
+  let hasArgAsSlot = false;
+  type InternalComponentApi = {
+    __docgenInfo: { props?: Record<string, { type: { name: string } }> };
+    [k: string]: unknown;
+  };
+  const component = story.component as InternalComponentApi;
+  const docGenProps = component?.__docgenInfo?.props;
+
+  if (!docGenProps) {
+    return component;
+  }
+
+  Object.entries(docGenProps).forEach(([key, argType]) => {
+    const value: string = argType?.type?.name;
+    if (value.includes('WithSlotShorthandValue')) {
+      hasArgAsSlot = true;
+      const match = value.match(slotRegex);
+      if (match) {
+        component.__docgenInfo.props![key].type.name = `Slot<\"${match[1]}\">`;
+        // @ts-expect-error - storybook doesn't ship proper types (value is missing)
+        updatedArgTypes[key].type.value = `Slot<\"${match[1]}\">`;
+      } else {
+        component.__docgenInfo.props![key].type.name = `Slot`;
+        // @ts-expect-error - storybook doesn't ship proper types (value is missing)
+        updatedArgTypes[key].type.value = `Slot`;
+      }
+    }
+  });
+
+  return { component, hasArgAsSlot, hasArgAsProp };
+}
+
+const AdditionalApiDocs: React.FC<{ children: React.ReactElement | React.ReactElement[] }> = ({ children }) => {
   const styles = useStyles();
+  return (
+    <div className={styles.additionalInfo}>
+      <div className={styles.additionalInfoMessage}>
+        <InfoFilled className={styles.additionalInfoIcon} />
+        <div className={styles.infoIcon}>{children}</div>
+      </div>
+    </div>
+  );
+};
+const RenderArgsTable = ({
+  hideArgsTable,
+  story,
+}: {
+  story: PrimaryStory;
+  hideArgsTable: boolean;
+  argTypes: StrictArgTypes;
```

Comment:
we dont use this - pls remove

```suggestion
  argTypes: StrictArgTypes;
```

Reply:
thanks, removed